pkg/auth/authenticator/password/ldappassword/util/server.go

```diff
@@ -0,0 +1,84 @@
+package ldaputil
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net"
+
+	"github.com/go-ldap/ldap"
+)
+
+// New returns a new LDAPServerConfig
+func New(url LDAPURL, insecure bool, tlsConfig *tls.Config, timeLimit int) LDAPServerConfig {
```

Comment:
Would you reasonably expect to be setting a timeout differently for different data fetch requests?